### PR TITLE
[FE-57] Image integration

### PR DESCRIPTION
**Pull request by dadamczykk**
https://www.loom.com/share/150a411c748e4fb5a19b29d576d199a3?sid=dd9f8d4e-7274-48ad-846e-5d3b12e32669

Because I'm kind of moron [I've pushed previous commit to the main](https://github.com/Jullija/oops/commit/6b4198c0c2d3511bf84120d92b339fb1a8739e07). Damage has been done. This pr reverts most of the done damage (so the removals of two files is good)

One file that is partially included in that pr, and needs to be fully checked is ```useUserPointsImage```. The rest is changed in this pr (maybe excluding some css properties, but who cares about them)

### CODE REVIEW

**Comment by AnCichocka** [comment]
(no one cares about css properties for now)

File: frontend/src/components/StudentProfile/userCard.tsx [hunk 1]
```diff
@@ -60,12 +60,14 @@ export function UserCard({ fullName, index, points }: UserCardProps) {
       <div style={styles.userImageContainer}>
         {loading && <div>Loading image...</div>}
         {error && <div>Error loading image</div>}
-        {imageId && !loading && !error && (
-          <ImageCache
-            imageId={imageId}
-            style={styles.userImageContainer}
-            imgStyle={styles.userImage}
+        {imageId && !loading && !error ? (
```
Comment (AnCichocka) [inline frontend/src/components/StudentProfile/userCard.tsx:63]:
```suggestion
        {displayImage ? (
```
please

File: frontend/src/components/StudentProfile/userCard.tsx [hunk 2]
```diff
@@ -60,12 +60,14 @@ export function UserCard({ fullName, index, points }: UserCardProps) {
...
+          <img
+            src={`${FILES_URL}${imageId}`}
+            alt={`Image id ${imageId}`}
+            style={styles.userImage}
           />
+        ) : (
+          !loading && !error && <div style={styles.userImageContainer} />
```
Comment (AnCichocka) [inline frontend/src/components/StudentProfile/userCard.tsx:70]: I would like it to be exported to variable and named properly like `showImageAltComponent`

File: frontend/src/components/hallOfFame/hallOfFameHeader.tsx [hunk 3]
```diff
@@ -12,7 +12,7 @@ const styles: Styles = {
     backgroundColor: "white",
     borderBottom: "1px solid #ccc",
     fontWeight: "bold",
-    zIndex: 100,
+    zIndex: 9,
```
Comment (AnCichocka) [inline frontend/src/components/hallOfFame/hallOfFameHeader.tsx:15]: Scary (joking)

File: frontend/src/hooks/useUserPointsImage.ts [hunk 4]
```diff
@@ -19,17 +19,18 @@ export const useUserPointsImage = (points: number) => {
     const level = data.editionByPk.levels.find((level) => {
       if (
         level.highest &&
-        points >
+        points >=
           (level.maximumPoints
             ? +level.maximumPoints
             : Number.POSITIVE_INFINITY)
       ) {
```
Comment (AnCichocka) [inline frontend/src/hooks/useUserPointsImage.ts:26]: see my comment - https://github.com/Jullija/oops/commit/6b4198c0c2d3511bf84120d92b339fb1a8739e07#r143854769

File: frontend/src/hooks/useUserPointsImage.ts [hunk 5]
```diff
@@ -19,17 +19,18 @@ export const useUserPointsImage = (points: number) => {
...
         return true;
       }
-      if (level.minimumPoints && points >= +level.minimumPoints) {
-        if (level.maximumPoints) {
-          return points <= +level.maximumPoints;
-        }
+      if (
+        level.minimumPoints &&
+        points >= +level.minimumPoints &&
+        (level.maximumPoints ? points < +level.maximumPoints : true)
```
Comment (AnCichocka) [inline frontend/src/hooks/useUserPointsImage.ts:32]: Just thinking if it's not more readable, but without difference to me

```suggestion
        && (!level.maximumPoints || points < +level.maximumPoints)
```

File: frontend/src/hooks/useUserPointsImage.ts [hunk 6]
```diff
@@ -19,17 +19,18 @@ export const useUserPointsImage = (points: number) => {
...
+      ) {
         return true;
       }
       return false;
```
Comment (AnCichocka) [inline frontend/src/hooks/useUserPointsImage.ts:36]: I believe you don't cover variant with first level?

Comment (AnCichocka) [reply frontend/src/hooks/useUserPointsImage.ts:36]: I'm not sure what `highest` in `level.highest` means, but if it's the highest level of all levels no objections

**Review by AnCichocka — CHANGES_REQUESTED** [review]
Please check if useUserPointsImage detects the smallest level. I also left some comments here: https://github.com/Jullija/oops/commit/6b4198c0c2d3511bf84120d92b339fb1a8739e07. Overall, looks good!

File: frontend/src/hooks/useUserPointsImage.ts [hunk 7]
```diff
@@ -16,20 +16,22 @@ export const useUserPointsImage = (points: number) => {
       return null;
     }
 
+    // TODO logic to find out which level is user
```
Comment (Szyntos) [inline frontend/src/hooks/useUserPointsImage.ts:19]: this is in a PR, so i didn't bother checking

**Review by AnCichocka — APPROVED** [review]
Hall of fame may look off after this merge, but don't worry, some files are deleted anyway in other PRs

**Pull request merged**